### PR TITLE
Fixing problems with watches evaluation on (not only) the upcoming JD…

### PR DESCRIPTION
**Pull request by jlahoda**
…K 11, fixing EvaluatorTest to pass on both JDK 8 with nb-javac and on recent JDK 11.

### CODE REVIEW

File: debugger.jpda.projects/src/org/netbeans/modules/debugger/jpda/projects/IntroduceClass.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ boolean computeIntroduceMethod(TreePathHandle h, CompilationInfo info, TreePath
                 lsBegin -= codeOffset;
                 this.snippetCode = new StringBuilder(this.snippetCode)
                         .insert((int) lsBegin, "return ")
+                        .append(';')
```
Comment (jlahoda) [inline debugger.jpda.projects/src/org/netbeans/modules/debugger/jpda/projects/IntroduceClass.java]: This is only important for vanilla javac, nb-javac's error recovery is able to handle the missing semi-colon. But having the semicolon should not do any harm.

File: debugger.jpda/nbproject/project.properties [hunk 2]
```diff
@@ -25,5 +25,6 @@ requires.nb.javac=true
 spec.version.base=1.108.0
 test-unit-sys-prop.test.dir.src=${basedir}/test/unit/src/
 test-unit-sys-prop.netbeans.user=${basedir}/work/nb_user_dir
+test.unit.cp.extra=../java.source.nbjavac/build/test-nb-javac/cluster/modules/org-netbeans-modules-java-source-nbjavac-test.jar
```
Comment (jlahoda) [inline debugger.jpda/nbproject/project.properties]: A trick to allow java.source.nbjavac to be enabled in tests when running tests with NB module system. Already used by refactoring.java, but will need tweaking in the future to simplify running the tests on vanilla javac as well.

File: debugger.jpda/src/org/netbeans/modules/debugger/jpda/expr/EvaluatorVisitor.java [hunk 3]
```diff
@@ -1992,7 +1992,7 @@ public Mirror visitIdentifier(IdentifierTree arg0, EvaluationContext evaluationC
                 identifierPath = getCurrentPath();
             }
             elm = evaluationContext.getTrees().getElement(identifierPath);
-            if (elm instanceof TypeElement && ((TypeElement) elm).asType() instanceof ErrorType) {
+            if (elm instanceof TypeElement && ((TypeElement) elm).asType().getKind() == TypeKind.ERROR) {
```
Comment (jlahoda) [inline debugger.jpda/src/org/netbeans/modules/debugger/jpda/expr/EvaluatorVisitor.java]: ClassTypes are both DeclaredType and ErrorType in current JDK 11. This presumably would be merged into nb-javac at some point.

File: debugger.jpda/test/unit/src/org/netbeans/api/debugger/jpda/JPDASupport.java [hunk 4]
```diff
@@ -241,7 +245,6 @@ public void doFinish () {
         debuggerEngine.getActionsManager ().
             doAction (ActionsManager.ACTION_KILL);
         waitState (JPDADebugger.STATE_DISCONNECTED);
-        deleteUserDir();
```
Comment (jlahoda) [inline debugger.jpda/test/unit/src/org/netbeans/api/debugger/jpda/JPDASupport.java]: Deleting the userdir at the end of the first test, while the module system is still running and some test execution is still pending does not work very well.

File: debugger.jpda/nbproject/project.properties (hunk 2 above)
Comment (JaroslavTulach) [reply debugger.jpda/nbproject/project.properties]: True, this is nasty.

File: nbjunit/src/org/netbeans/junit/NbModuleSuite.java [hunk 5]
```diff
@@ -507,6 +507,19 @@ public Configuration reuseUserDir(boolean reuse) {
             ,honorAutoEager, failOnMessage, failOnException, hideExtraModules);
         }
 
+        /**
+         * Sets the parent ClassLoader on which the NB platform should start.
+         * @param parentCL the parent ClassLoader
+         * @return clone of this configuration with the parent ClassLoader set
+         * @since 1.90
+         */
+        public Configuration parentClassLoader(ClassLoader parentCL) {
```
Comment (JaroslavTulach) [inline nbjunit/src/org/netbeans/junit/NbModuleSuite.java]: OK.

**Pull request merged**